gamestonk_terminal/fred/fred_api.py

```diff
@@ -0,0 +1,143 @@
+from fredapi import Fred
+import argparse
+import matplotlib.pyplot as plt
+from gamestonk_terminal.helper_funcs import (
+    parse_known_args_and_warn,
+    valid_date,
+    str_to_bool,
+)
+from gamestonk_terminal.config_terminal import API_FRED_KEY
+
+
+def get_GDP(l_args):
+    fred = Fred(api_key=API_FRED_KEY)
+    parser = argparse.ArgumentParser(
+        add_help=False,
+        prog="GDP",
+        description="""
+                GDP
+            """,
+    )
+    parser.add_argument(
+        "-n",
+        dest="n_to_get",
+        type=int,
+        default=-1,
+        required=False,
+        help="Number of GDP Values to Grab",
+    )
+
+    parser.add_argument(
+        "-s",
+        dest="start_date",
+        type=valid_date,
+        default="2015-01-01",
+        required=False,
+        help="Date to Start",
+    )
+
+    parser.add_argument(
+        "-p",
+        dest="plot",
+        type=str_to_bool,
+        default=True,
+        required=False,
+        help="Plot GDP",
+    )
+
+    try:
+        ns_parser = parse_known_args_and_warn(parser, l_args)
+
+        if not ns_parser:
+            return
+
+        gdp = fred.get_series("GDP", ns_parser.start_date)
+
+        if int(ns_parser.n_to_get) > 0:
+            lastn = gdp.tail(int(ns_parser.n_to_get))
+
+        else:
+            lastn = gdp
+
+        for date, val in lastn.iteritems():
+            print(f'Date: {date.strftime("%m-%d-%Y")}, GDP: {val} ')
+
+        if ns_parser.plot:
+            plt.figure()
+            lastn.plot(style="-ok")
+            plt.xlabel("Time")
+            plt.ylabel("GDP")
+            plt.show()
+
+    except Exception as e:
+        print(e)
+        print("")
+        return
+
+
+def custom_data(l_args):
+    fred = Fred(api_key=API_FRED_KEY)
+
+    parser = argparse.ArgumentParser(
+        add_help=False,
+        prog="Custom",
+        description="""
+                    Custom Data
+                """,
+    )
+    parser.add_argument(
+        "-id", dest="series_id", required=True, type=str, help="FRED Series ID"
+    )
+
+    parser.add_argument(
+        "-s",
+        dest="start_date",
+        type=valid_date,
+        default="2015-01-01",
+        required=False,
+        help="Date to Start",
+    )
+    parser.add_argument(
```

Comment:
Given that the data will never be much, I think the plot and disp arguments are unnecessary.

Also, you were parsing this in an inefficient way. Because instead of having to do:
`-p true` or `-p false` you can set a flag to have the value you want.

E.g. If the user doesn't add anything you display a plot by default, if in the command he does something like:
```
gdp --noplot
```
Then we recognize that `--noplot` is a flag and we set that to the value we want.

This is done using:
```
parser.add_argument(
        "-q",
        "--quarter",
        action="store_true",
        default=False,
        dest="b_quarter",
        help="Quarter fundamental data flag.",
    )
```
Notice the default to False, but if the user sets this flag ('store_true') that value will be True instead.

Anyway, for these functions I would just keep plots and prints a default